invenio_rdm_records/services/communities/service.py

```diff
@@ -241,6 +246,19 @@ def remove(self, identity, id_, data, uow):
             raise_errors=True,
         )
         communities = valid_data["communities"]
+
+        # Run components
+        for component in self.components:
+            if hasattr(component, "remove"):
```

Comment:
I would move this to `_remove` after the permission checks and rename to `remove_community`. Also, the question on where to add the component hook is valid here too.